core/services/pipeline/task.eth_call.go

```diff
@@ -62,7 +70,21 @@ func (t *ETHCallTask) Run(ctx context.Context, vars Vars, inputs []Result) (resu
 
 	resp, err := chain.Client().CallContract(ctx, call, nil)
 	if err != nil {
+		if t.ExtractRevertReason {
+			err = t.retrieveRevertReason(err)
+		}
+
 		return Result{Error: err}
```

Comment:
this err should be a combination of err (line 72) and err (line 74). Could we maybe do something like...

```
err = t.withRevertReason(err)

func (t *ETHCallTask) retrieveRevertReason(baseErr error) error {
	reason, err := eth.ExtractRevertReasonFromRPCError(baseErr)
	// ...
	return errors.Wrap(baseErr, reason)
}
```

Reply:
Good idea, done